nssp_demo/prep_data.py

```diff
@@ -181,20 +116,150 @@
         "data_observed_hospital_admissions": train_ed_admissions,
         "test_ed_admissions": test_ed_admissions,
         "state_pop": state_pop,
+        "right_truncation_offset": right_truncation_offset
     }
 
-    model_folder_name = f"{disease.lower()}_r_{report_date}_f_{actual_first_date}_l_{actual_last_date}_t_{last_training_date}"
-
-    model_folder = pathlib.Path("private_data", model_folder_name)
-    os.makedirs(model_folder, exist_ok=True)
-    state_folder = pathlib.Path(model_folder, state_abb)
-    os.makedirs(state_folder, exist_ok=True)
-    logger.info(f"Saving {state_abb}")
-    data_to_save.to_csv(str(pathlib.Path(state_folder, "data.csv")))
+    state_dir = os.path.join(model_data_dir, state_abb)
+    os.makedirs(state_dir, exist_ok=True)
+    if logger is not None:
+        logger.info(f"Saving {state_abb} to {state_dir}")
+    data_to_save.to_csv(str(pathlib.Path(state_dir, "data.csv")))
 
-    with open(
-        pathlib.Path(state_folder, "data_for_model_fit.json"), "w"
-    ) as json_file:
+    with open(os.path.join(state_dir, "data_for_model_fit.json"), "w"
+              ) as json_file:
         json.dump(data_for_model_fit, json_file)
 
-logger.info("Data preparation complete.")
+
+def main(disease,
```

Comment:
We can either get rid of `main` and warp its contents in  `if __name__ == "__main__": ` or just wrap the call to `main`.